File: javascript/series/series.js
function Series(string) {
	this.digits = digitize(string);
}

Series.prototype.slices = function(slicesNum) {
	if (slicesNum > this.digits.length) {
		throw new Error('Slice size is too big.');
	}
	
	var limit = this.digits.length - (slicesNum - 1);
	var generatedSlices = [];
	
	for (var i = 0; i < limit; ++i) {
		var slice = [];
		for (var span = 0; span < slicesNum; ++span) {
			slice.push(this.digits[i + span]);
		}
		
		generatedSlices.push(slice);
	}

	return generatedSlices;
};

function digitize(string) {
	var digits = [];
	var stringLength = string.length;
	for (var i = 0; i < stringLength; ++i) {
		digits.push(parseInt(string[i]));
	}
	
	return digits;
}

module.exports = Series;

File: python/series/series.py
def slices(serie, slices_num):
	if (slices_num > len(serie)) or (slices_num < 1):
		raise ValueError('Slice size is too big.')
	
	limit = len(serie) - (slices_num - 1)
	generated_slices = []
	
	for i in range(limit):
		current_slice = []
		for span in range(slices_num):
			current_slice.append(int(serie[i + span]))
		
		generated_slices.append(current_slice)

	return generated_slices